### PR TITLE
python3: no filter and no xrange

### PR DESCRIPTION
**Pull request by fchapoton**
replaced all "in xrange" by "in range"

fix the two wrong calls to filter

Sage 9.0 is out and uses Python 3...

### CODE REVIEW

File: abelfunctions/singularities.py [hunk 1]
```diff
@@ -170,13 +170,11 @@ def has_finite_v(Pi):
             n,alpha = zip(*Pi.terms)
 
             # if there are still no terms then they are positive exponent
-            if n == []:
+            if not n:
                 return True
-            elif min(n) >= 0:
-                return True
-            return False
+            return min(n) >= 0:
```
Comment (rparini) [inline abelfunctions/singularities.py]: Syntax error here
```suggestion
            return min(n) >= 0
```

**Comment by fchapoton** [comment]
I fixed the syntax error.

**Comment by fchapoton** [comment]
many other things have to be fixed to reach python3 compatibility, see the result of

pylint --py3k -d W1618,W1619 abelfunctions/

File: abelfunctions/singularities.py [hunk 2]
```diff
@@ -170,13 +170,11 @@ def has_finite_v(Pi):
...
-            if n == []:
+            if not n:
                 return True
-            elif min(n) >= 0:
-                return True
-            return False
+            return min(n) >= 0
 
-        P = filter(has_finite_v, P)
+        P = [has_finite_v(elt) for elt in P]
```
Comment (rparini) [inline abelfunctions/singularities.py]:
```suggestion
        P = [elt for elt in P if has_finite_v(elt)]
```

**Pull request merged**